File: simulator/include/Configuration.php
<?php
namespace PLUSPEOPLE\Pesapi\simulator;

set_include_path("/Users/kaal/work/2011/PesaPi/simulator/include:" . get_include_path());
date_default_timezone_set('Africa/Nairobi');



class Configuration {
	static public $singleton = null;

	protected $configArray=array(
															 // Database settings follow - please note that they are repeated twice
															 "DatabaseHostRead"						=> "localhost",
															 "DatabaseUserRead"						=> "root",
															 "DatabasePasswordRead"				=> "ture",
															 "DatabaseDatabaseRead"				=> "2011_simulator",
															 "DatabaseHostWrite"						=> "localhost",
															 "DatabaseUserWrite"						=> "root",
															 "DatabasePasswordWrite"				=> "ture",
															 "DatabaseDatabaseWrite"				=> "2011_simulator",
															 );

	/////////////////////////////////////////////
	public function getConfig($argument) {
		return $this->configArray[$argument];
	}

	public static function instantiate() {
		if (self::$singleton == null) {
			self::$singleton = new Configuration();
		}
		return self::$singleton;
	}
}

?>

File: csharp/source/Configuration.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace PLUSPEOPLE.Pesapi
{
	public class Configuration
	{
		private static Configuration singleton ;
		private Dictionary<string, string> configs = new Dictionary<string, string>();

		private Configuration()
		{
			configs.Add("ProductionMode", "off");
			configs.Add("SimulationMode", "on");
			configs.Add("AllowAutoUpdate", "on");
			configs.Add("AllowFeedback", "on");
			configs.Add("MaxCompatibility", "on");
			configs.Add("MpesaCertificatePath", "");
			configs.Add("MpesaLoginName", "");
			configs.Add("MpesaCorporation", "");
			configs.Add("MpesaInitialSyncDate", "");
			configs.Add("CookieFolderPath", "");
			configs.Add("DatabaseHostRead", "localhost");
			configs.Add("DatabaseUserRead", "");
			configs.Add("DatabasePasswordRead", "");
			configs.Add("DatabaseDatabaseRead", "");
			configs.Add("DatabaseHostWrite", "localhost");
			configs.Add("DatabaseUserWrite", "");
			configs.Add("DatabasePasswordWrite", "");
			configs.Add("DatabaseDatabaseWrite", "");
			configs.Add("Version", "0.0.3");
		}


		public string GetConfig(string configName)
		{
			return this.configs[configName];
		}

		public void SetConfig(string configName, string value)
		{
			if (this.configs.ContainsKey(configName))
				{
					this.configs[configName] = value;
				} 
			else
				{
					this.configs.Add(configName, value);
				}
		}

		public static Configuration Instantiate()
		{
			if (null == Configuration.singleton)
				{
					Configuration.singleton = new Configuration();
				}
			return Configuration.singleton;
		}
	}
}
